File: TheAnvil/src/main/java/com/qualityplus/anvil/api/box/Box.java
package com.qualityplus.anvil.api.box;

import com.qualityplus.anvil.api.config.ConfigFiles;
import com.qualityplus.anvil.base.config.*;
import org.bukkit.plugin.Plugin;

public interface Box {
    ConfigFiles<Config, RecipesFile, Inventories, Messages, Commands> files();
    Plugin plugin();
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/api/box/Box.kt
package com.qualityplus.anvil.api.box

import com.qualityplus.anvil.api.config.ConfigFiles
import com.qualityplus.anvil.base.config.Commands
import com.qualityplus.anvil.base.config.Config
import com.qualityplus.anvil.base.config.Inventories
import com.qualityplus.anvil.base.config.Messages
import org.bukkit.plugin.Plugin

interface Box {
    val files: ConfigFiles<Config, Inventories, Messages, Commands>
    val plugin: Plugin
}